FinalSoftdesProject.py

```diff
@@ -0,0 +1,162 @@
+pygame.init()
+
+
+class Text:
+
+    """
+        This class makes text in the window. It is called in the Game class.
+        It can refer to text within the buttons or above them.
+    """
+    def __init__(self, text='', color=(255, 255, 255), size=(150, 450, 100, 50)):
+        # takes in text and color from Game class
+        self.text = text
+        self.color = color
+        self.size = size
+        self.clock = pygame.time.Clock()
+
+    def message(self, font):
+        # Defines the surface the text is written on.
+        textSurface = font.render(self.text, True, self.color)
+        return textSurface, textSurface.get_rect()
+
+    def message_display(self, gameDisplay):
+        # Displays message with font, size, text, and then pauses half a sec.
+        largeText = pygame.font.SysFont("monospace", 20)
+        TextSurf, TextRect = self.message(largeText)
+        gameDisplay.blit(TextSurf, TextRect)
+        pygame.display.update()
+        time.sleep(2)  # How long game pauses before next step appears
+
+    def button_text_display(self, gameDisplay):
+        x = self.size[0]
+        y = self.size[1]
+        w = self.size[2]
+        h = self.size[3]
+        smallText = pygame.font.SysFont("monospace", 40)
+        textSurf, textRect = self.message(smallText)
+        textRect.center = (x+(w/2), y+(h/2))
+        gameDisplay.blit(textSurf, textRect)
+        pygame.display.update()
+        # self.clock.tick(1)
+
+class Button:
+    """
+    This defines the buttons that appear in the game. The buttons are called in
+    the Game cass and take in the button sizes, color, and color the button
+    changes to when the mouse hovers over it.
+    """
+    def __init__(self, size=(150, 450, 100, 50), buttontext='', color=(200, 0, 0),
+                 color_bright=(255, 0, 0)):
+        # initial values are defined in case the program doesn't. These values
+        # are updated from the Gane class with self.size = size, etc.
+        self.size = size
+        self.color = color
+        self.white = (255, 255, 255)
+        self.gameDisplay = pygame.display.set_mode((800, 600))
+        self.color_bright = color_bright
+        self.mouse = pygame.mouse.get_pos()
+        self.click = pygame.mouse.get_pressed()
+
+    def make_rectangle(self, gameDisplay):
+        # Draws the rectangle that defines the button with a color and size
+        # pulled from itself.
+        x = self.size[0]
+        y = self.size[1]
+        w = self.size[2]
+        h = self.size[3]
+        mouse = pygame.mouse.get_pos()
+        if x+w > mouse[0] > x and y+h > mouse[1] > y:
+            pygame.draw.rect(gameDisplay, self.color_bright, self.size)
+        else:
+            pygame.draw.rect(gameDisplay, self.color, self.size)
+
+    def button_action(self, gameDisplay, action=None):
+        self.click = pygame.mouse.get_pressed()
+        mouse = pygame.mouse.get_pos()
+        mouse_right_of_left_side = self.size[0] < mouse[0]
```

Comment:
These variable names are a bit too long; they don't need to be super descriptive, just memorable.